Update Shortest paths docs

nhz2 opened the pull request:
I added some details to the Shortest paths section of the docs.

Maybe some of the specific `dijkstra_shortest_paths` info should be moved to `function dijkstra_shortest_paths` docs?

nhz2 commented:
@etiennedeg  @gdalle , could you please review this?

I would like to write an edge betweenness function using the `dijkstra_shortest_paths` in  `Graphs.jl`, but its outputs are currently poorly documented.

etiennedeg commented on docs/src/first_steps/paths_traversal.md:
```diff
@@ -31,7 +31,7 @@ The package also includes uniform random walks and self avoiding walks with the
 
 - The distance from a vertex to itself is always `0`.
-- The distance between two vertices with no connecting edge is always `Inf`.
+- The distance between two vertices with no connecting edge is always `Inf` or `typemax(Int64)`.
```
```suggestion
- The distance between two vertices with no connecting edge is always `Inf` or `typemax(eltype(distmx))`.
```

etiennedeg reviewed (commented):
Thanks for your contribution, I think we should also document this in [Algorithms API](https://juliagraphs.org/Graphs.jl/dev/algorithms/shortestpaths/), this is actually less documented than the [Paths and traversal](https://juliagraphs.org/Graphs.jl/dev/first_steps/paths_traversal/) section.
As a side note, we should also add accessors for the `AbstractPath` object, we should not need to rely on internal fields.

nhz2 commented:
> Thanks for your contribution, I think we should also document this in [Algorithms API](https://juliagraphs.org/Graphs.jl/dev/algorithms/shortestpaths/), this is actually less documented than the [Paths and traversal](https://juliagraphs.org/Graphs.jl/dev/first_steps/paths_traversal/) section. As a side note, we should also add accessors for the `AbstractPath` object, we should not need to rely on internal fields.

Thanks for the review, I moved the `dijkstra_shortest_paths` specific info to the `dijkstra_shortest_paths` docstring,
and added links to the referenced functions in [Paths and traversal](https://juliagraphs.org/Graphs.jl/dev/first_steps/paths_traversal/).

gdalle commented:
Thanks @etiennedeg
